web/src/cypress/support/schedule.ts

```diff
@@ -138,6 +145,56 @@ function deleteSchedule(id: string): Cypress.Chainable<void> {
   })
 }
 
+function createTemporarySchedule(
+  scheduleID?: string,
+  options?: Partial<TemporarySchedule> & { scheduleID?: string },
+): Cypress.Chainable<void> {
+  const mutation = `
+    mutation($input: SetTemporaryScheduleInput!) {
+      setTemporarySchedule(input: $input)
+    }
+  `
+
+  if (!scheduleID) {
+    return cy
+      .createSchedule()
+      .then((s: Schedule) => createTemporarySchedule(s.id, options))
+  }
+
+  const nowDT = DateTime.local()
+  const input = options || {}
+  input.scheduleID = scheduleID
+
+  // set start to start of today or 7 days before set end
+  if (!input.start && !input.end) {
+    input.start = nowDT.startOf('day').toISO()
+  } else if (!input.start && input.end) {
+    input.start = DateTime.fromISO(input.end).minus({ days: 7 }).toISO()
+  }
+
+  // set end to 7 days after start
+  if (!input.end) {
+    const start = DateTime.fromISO(input.start as string)
+    input.end = start.plus({ days: 7 }).endOf('day').toISO()
+  }
+
+  // set a single shift to extend entire fixed shift duration
+  if (!input.shifts?.length) {
+    cy.fixture('users').then((users) => {
+      input.shifts = [
+        {
+          start: input.start as string,
+          end: input.end as string,
+          userID: users[1].id,
+        } as OnCallShift,
+      ]
```

Comment:
let's randomize this so tests don't depend on consistency

Reply:
I wasn't sure how to best approach this — since we only have 3 users in our fixture I needed to be somewhat strict since I'll be utilizing all three (1 for the original assignment on create, 1 to add in form, and 1 for misc if needed), so making it random could cause inconsistencies if we're not really careful. However if you still think that's the best way to approach it I'll go ahead and see what I can do 🙂

Reply:
From the support function, everything not explicitly provided should be random, including the number of shifts and their users.

If a test has any requirement on the data used (e.g. like a single shift for a specific user), that should be defined in the test itself. Otherwise, a future change here could break an unknown number of tests.